contracts/utilities/TokenFeeCollector.sol

```diff
@@ -0,0 +1,150 @@
+        uint payeeIndex = tInfo.payees.length+1;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            if (tInfo.payees[i] == devAddress) {
+                payeeIndex = i;
+                break;
+            }
+        }
+        require(payeeIndex < tInfo.payees.length, "Payee not found");
+        _removeElement(tokenAddress, payeeIndex);
+    }
+
+    function editPayee(address tokenAddress, address devAddress, uint paymentBips) external onlyOwner {
+        require(
+            tokenPayees[tokenAddress].isValue,
+            "This token is not registered, use registerToken to initialize it"
+        );
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        uint payeeIndex = tInfo.payees.length+1;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            if (tInfo.payees[i] == devAddress) {
+                payeeIndex = i;
+                break;
+            }
+        }
+        require(payeeIndex < tInfo.payees.length, "Payee not found");
+        tInfo.paymentBips[payeeIndex] = paymentBips;
+    }
+
+    function viewPayee(address tokenAddress, uint index) external view onlyOwner returns(address, uint) {
+        require(
+            tokenPayees[tokenAddress].isValue,
+            "The token is not registered, use registerToken to initialize it"
+        );
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        require(index < tInfo.payees.length, "You're reading from outside the boundaries");
+        return (tInfo.payees[index], tInfo.paymentBips[index]);
+    }
+
+    function collectFee(address tokenAddress) external {
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        uint payeeIndex = tInfo.payees.length+1;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            if (tInfo.payees[i] == msg.sender) {
+                payeeIndex = i;
+                break;
+            }
+        }
+        require(payeeIndex < tInfo.payees.length || msg.sender == owner(), "You're not a payee or the owner, don't collect");
+        uint totalBips = 0;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            totalBips += tInfo.paymentBips[i];
+        }
+
+        require(totalBips <= maxBips, "The total payout configured exceeds 100%, contact the feeCollect owner");
+        uint balance = IERC20(tokenAddress).balanceOf(address(this));
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            address payee = tInfo.payees[i];
+            uint paymentBips = tInfo.paymentBips[i];
+            IERC20(tokenAddress).transfer(payee, balance.mul(paymentBips).div(maxBips));
+        }
+    }
+
+    function viewBalance(address tokenAddress) external view returns (uint, uint) {
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        uint payeeIndex = tInfo.payees.length+1;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            if (tInfo.payees[i] == msg.sender) {
```

Comment:
`msg.sender` should be replaced with parameter `address account`

Reply:
the purpose was to only allow the owner of the private key to view the balance, but I definitely add an extra parameter to allow anyone view the balance of anyone